api/mfx_dispatch/linux/mfxloader.cpp

```diff
@@ -46,7 +46,7 @@ namespace MFX {
     #else
         #define LIBMFXSW "libmfxsw32.so.1"
         #define LIBMFXHW "libmfxhw32.so.1"
-        #define ONEVPLRT "libmfx-gen.so.1"
+        #define ONEVPLRT "libmfx-gen.so"
```

Comment:
Well, yes, this is a problem. However, I would argue that: 1) .so is a linker name and should not be used in dlopen() calls, 2) libmfx-gen should be fixed to have proper .1 soname instead of 1.2.

Also, you have CI failing on this pr...

Reply:
Is libmfxhw64.so not a linker name? libmfxhw64.so is used for MSDK RT

Reply:
OneVPL dispatcher uses "libmfx-gen.so.1.2".  The second number is the "mfx major version".

Reply:
... that is, it is the API major version.

Reply:
We will have to change libmfx code in the future if using libmfx-gen.so.1.2 after bumping a new major version for libvpl

Reply:
It's unclear to me why "1" is the prefix.

Reply:
The first number should be "major"

Reply:
In current version scheme, if mfx major changes but not library major, then we expect it's still backwards compatible?

Reply:
@dvrogozh I don't understand why this change cause CI failing, maybe this is a case issue. Is there a way to run CI locally?

Reply:
> We will have to change libmfx code in the future if using libmfx-gen.so.1.2 after bumping a new major version for libvpl

OneVPL dispatcher has the same problem.

Reply:
Yes, of course, just run `make test`.

Reply:
> In current version scheme, if mfx major changes but not library major, then we expect it's still backwards compatible?

If so, then @dvrogozh original #2 should be the fix.  But also would require vpl dispatcher fix.
If not, then the "1." should be dropped in the libmfx-gen version.  And still requires vpl dispatcher fix, too.

The only way to avoid vpl dispatcher change is to load libmfx-gen.so.1.2 in mfx dispatcher.

Reply:
However, I think @dvrogozh suggested fix is the "right" fix.  And, we will need to do some damage control with vpl dispatcher.

Reply:
@jonrecker any comment for oneVPL library name?